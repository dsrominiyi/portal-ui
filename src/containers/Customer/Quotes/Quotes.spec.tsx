import React from 'react';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { shallow, mount, ShallowWrapper } from 'enzyme';

import ConnectedQuotes, { Quotes } from '.';

import mockData from '../../../../__mocks__/mock-data.json';

const mockStore = createMockStore([]);

const initialProps = {
  customer: mockData.customers[0],
  quotes: mockData.quotes
};

let container: ShallowWrapper;
const connectedContainer = mount(
  <Provider store={mockStore({ customer: initialProps })}>
    <ConnectedQuotes />
  </Provider>
);

const render = (props: any = initialProps) => {
  container = shallow(<Quotes {...props} />);
};

describe('Customer', () => {
  beforeEach(() => {
    render();
  });

  test('the default export is a connected component', () => {
    const mappedProps = connectedContainer.find('Quotes').props();

    expect(mappedProps).toEqual(initialProps);
  });

  it('matches snapshot for customer quotes', () => {
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot for a company customer', () => {
    render({ ...initialProps, customer: mockData.customers[1] });

    expect(container).toMatchSnapshot();
  });

  test('matches snapshot with no quotes associated', () => {
    render({ ...initialProps, quotes: [] });

    expect(container).toMatchSnapshot();
  });

  test('HandleNewQuote function is triggered when New Quote button is clicked', () => {
    const button = container.find('Button').at(0);
    const event = { preventDefault: jest.fn() };
    window.alert = jest.fn();
    button.simulate('click', event);

    expect(event.preventDefault).toHaveBeenCalled();
  });
});
