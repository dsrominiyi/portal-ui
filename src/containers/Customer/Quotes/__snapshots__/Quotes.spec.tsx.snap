// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer matches snapshot for a company customer 1`] = `
<div
  className="pageContainer"
>
  <div
    className="topContainer"
  >
    <div
      className="titleContainer"
    >
      <PageTitle
        text="Quotes"
      />
    </div>
    <div
      className="actionsContainer"
    >
      <Button
        label="New Quote"
        onClick={[Function]}
        status="Success"
      />
    </div>
  </div>
  <div>
    <InfoBox
      additionalClasses={
        Array [
          "fullWidth",
        ]
      }
      header="Datamatters (Company)"
      items={
        Array [
          <ExtendedCustomerQuote
            quote={
              Object {
                "id": "12345",
                "insurer": "We Insure Any Car",
                "rate": 392.63,
                "salesExecutive": Object {
                  "name": Object {
                    "forename": "James",
                    "middleName": "Roland",
                    "surname": "Fowler",
                  },
                  "profilePhoto": "/images/person-who-does-not-exist.jpg",
                  "username": "Jimmy123",
                },
                "startDate": "2019-12-01",
                "type": "pc",
              }
            }
          />,
          <ExtendedCustomerQuote
            quote={
              Object {
                "id": "12346",
                "insurer": "I M Xpensive",
                "rate": 1485.21,
                "salesExecutive": Object {
                  "name": Object {
                    "forename": "John",
                    "middleName": "Robert",
                    "surname": "Tucker",
                  },
                  "profilePhoto": "/images/person-who-does-not-exist.jpg",
                  "username": "JohnABC",
                },
                "startDate": "2019-12-01",
                "type": "pc",
              }
            }
          />,
          <ExtendedCustomerQuote
            quote={
              Object {
                "decline": "Unnaceptable CCJs",
                "id": "12347",
                "insurer": "ABC Insurance",
                "rate": 0,
                "salesExecutive": Object {
                  "name": Object {
                    "forename": "Connor",
                    "middleName": "Adam",
                    "surname": "Robertson",
                  },
                  "username": "Connor123",
                },
                "startDate": "2019-12-01",
                "type": "pc",
              }
            }
          />,
        ]
      }
      useDivider={true}
    />
  </div>
</div>
`;

exports[`Customer matches snapshot for customer quotes 1`] = `
<div
  className="pageContainer"
>
  <div
    className="topContainer"
  >
    <div
      className="titleContainer"
    >
      <PageTitle
        text="Quotes"
      />
    </div>
    <div
      className="actionsContainer"
    >
      <Button
        label="New Quote"
        onClick={[Function]}
        status="Success"
      />
    </div>
  </div>
  <div>
    <InfoBox
      additionalClasses={
        Array [
          "fullWidth",
        ]
      }
      header="Mr John James Smith (Individual)"
      items={
        Array [
          <ExtendedCustomerQuote
            quote={
              Object {
                "id": "12345",
                "insurer": "We Insure Any Car",
                "rate": 392.63,
                "salesExecutive": Object {
                  "name": Object {
                    "forename": "James",
                    "middleName": "Roland",
                    "surname": "Fowler",
                  },
                  "profilePhoto": "/images/person-who-does-not-exist.jpg",
                  "username": "Jimmy123",
                },
                "startDate": "2019-12-01",
                "type": "pc",
              }
            }
          />,
          <ExtendedCustomerQuote
            quote={
              Object {
                "id": "12346",
                "insurer": "I M Xpensive",
                "rate": 1485.21,
                "salesExecutive": Object {
                  "name": Object {
                    "forename": "John",
                    "middleName": "Robert",
                    "surname": "Tucker",
                  },
                  "profilePhoto": "/images/person-who-does-not-exist.jpg",
                  "username": "JohnABC",
                },
                "startDate": "2019-12-01",
                "type": "pc",
              }
            }
          />,
          <ExtendedCustomerQuote
            quote={
              Object {
                "decline": "Unnaceptable CCJs",
                "id": "12347",
                "insurer": "ABC Insurance",
                "rate": 0,
                "salesExecutive": Object {
                  "name": Object {
                    "forename": "Connor",
                    "middleName": "Adam",
                    "surname": "Robertson",
                  },
                  "username": "Connor123",
                },
                "startDate": "2019-12-01",
                "type": "pc",
              }
            }
          />,
        ]
      }
      useDivider={true}
    />
  </div>
</div>
`;

exports[`Customer matches snapshot with no quotes associated 1`] = `
<div
  className="pageContainer"
>
  <div
    className="topContainer"
  >
    <div
      className="titleContainer"
    >
      <PageTitle
        text="Quotes"
      />
    </div>
    <div
      className="actionsContainer"
    >
      <Button
        label="New Quote"
        onClick={[Function]}
        status="Success"
      />
    </div>
  </div>
  <div>
    <InfoBox
      additionalClasses={
        Array [
          "fullWidth",
        ]
      }
      header="Mr John James Smith (Individual)"
      items={Array []}
      useDivider={true}
    />
  </div>
</div>
`;
